_automate/publish.sh

```diff
@@ -4,11 +4,22 @@ set -eu
 
 ORDER=(core server-utils tcp ws http ipc stdio pubsub core-client/transports core-client derive test)
 
-# First display the plan
-for crate in ${ORDER[@]}; do
-	cd $crate > /dev/null
+function read_toml () {
+	NAME=""
+	VERSION=""
+	NAME=$(grep "^name" ./Cargo.toml | sed -e 's/.*"\(.*\)"/\1/')
 	VERSION=$(grep "^version" ./Cargo.toml | sed -e 's/.*"\(.*\)"/\1/')
-	echo "$crate@$VERSION"
+}
+function remote_version () {
+	REMOTE_VERSION=""
+	REMOTE_VERSION=$(cargo search "$NAME" | grep "^$NAME =" | sed -e 's/.*"\(.*\)".*/\1/')
```

Comment:
`'s/.*"\(.*\)".*/\1/'` doesn't handle `alpha`/`beta` crates but I don't think we care (I also don't know if those even show up in `cargo search`?).

Another nit would be pathological cases such as `cargo search rust` (18676 matches): only the first 10 are displayed (no idea what ordering criteria, maybe the same as crates.io?) so if the one you're publishing is further down the list you'd need to use `--limit <some int>`. Again, I don't think that is important but you asked for a review so… ;)

Reply:
Well, we are limited to our `jsonrpc-` prefixed stuff, so I think it's good enough for our use case:)

Reply:
I was thinking ahead about adopting this in `parity-common` but maybe I should wait until it reaches the AI Singularity or at least Turing completness... ;)